### ISSUE TITLE
SvelteKit support

### ISSUE THREAD

**Issue by 24jr** [aws-amplify/amplify-js#9224]
### Is this related to a new or existing framework?

New framework

### Describe the feature you'd like to request

I along with many others are running into some issues trying to use sveltekit with aws amplify. I'm not sure if there is a formal feature request to get support for it or if it is being worked on to be supported by aws amplify. 

### Describe the solution you'd like

Ui components could be good eventually but even just getting it to work more easily or have simple guide would be great. I've managed to get around many of the gotchas but can't get it to build. Even if I somehow figure this part out it seams like there should be a more guided approach for using sveltekit with aws amplify

### Describe alternatives you've considered

https://github.com/aws/aws-sdk-js/issues/3673
https://github.com/aws-amplify/amplify-ui/issues/268
https://github.com/aws-amplify/amplify-console/issues/1298
https://github.com/aws-amplify/amplify-console/issues/2318
https://github.com/aws-amplify/amplify-js/issues/9056

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by ashika01** [comment]
@24jr Could you detail us some of the workaround you did and why you have to do it? Right now, we are also waiting on https://github.com/aws/aws-sdk-js/issues/3673. (Side note: Erik who opened up the issue is from our team). But if you have other things you think can be done in amplify that can unblock svelte usage, we would like to learn more of this.

**Comment by 24jr** [comment]
Yeah that one was one I looked at and I don't want to repeat stuff of course but not sure if there was really a sveltekit focused issue to find solution but rather a mish mash of a bunch of things.  So for me this is where I'm at. I'll just go over everything best I can remember

Standard skeleton svelte-kit project https://kit.svelte.dev/
```
npm init svelte@next my-app
cd my-app
npm install
```
I used aws amplify admin ui to create my backend so ran (of course this first `npm install -g @aws-amplify/cli`)
```
amplify pull --appId d2zuoroq5jzalo --envName staging
```
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using none
? Source Directory Path:  src
? Distribution Directory Path: dist
? Build Command:  npm run-script build
? Start Command: npm run-script start
? Do you plan on modifying this backend? No
Pretty much just kept the defaults for all. of course no sveltekit option for js framework so kept as javascript. However the last choice was important to say No for plan to modify. because if I say yes it fails with 
```
? Do you plan on modifying this backend? Yes
✔ Successfully pulled backend environment staging from the cloud.
✖ There was an error initializing your environment.
Failed to pull the backend.
Must use import to load ES Module: /Users/24jr/Documents/projects/2021/svelte/agility-all/agility/src/aws-exports.js
/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-frontend-javascript/lib/frontend-config-creator.js:1
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/24jr/Documents/projects/2021/svelte/agility-all/agility/src/aws-exports.js
```
I don't need to update the backend directly in the app so this was not a big deal. Selecting No was fine and worked. When I wanted update the graphQL schema lets say I would go to an empty folder outside of sveltekit proj and do the same `amplify pull --appId d2zuoroq5jzalo --envName staging` and say yes in that and it would succeed and I could make changes to schema from there. I didnt do anything other than update schema like this.

To get amplify to work in the app I added to app.html
```
<script>
  const isBrowser = () => typeof window !== 'undefined';
  const isGlobal = () => typeof global !== 'undefined';
  var exports = {};
  if (!isGlobal() && isBrowser()) {
      var global = window;
  }
</script>
```
I put the Amplify imports in the __layout.svelte route because my understanding is it's a main part of every page and context="module" runs first
```
<script context="module">
  import "../app.css";
  import Layout from "$lib/views/Layout/index.svelte";
  // import Amplify from "aws-amplify";
  import { Amplify } from '@aws-amplify/core'
  import awsExports from "../aws-exports";
  Amplify.configure({ ...awsExports, ssr: true });
  import { initAuth } from "$lib/components/Auth/store";
  initAuth();
</script>
``` 
I believe was having some issue with `import Amplify from "aws-amplify";` however `import { Amplify } from '@aws-amplify/core'` worked as alternative

From this point everything worked fine in the app with authentication and making mutations and queries to my backend as usually done with API.graphql(... 

To build I used adapter-static from https://github.com/sveltejs/kit/tree/master/packages/adapter-static
```
npm i -D @sveltejs/adapter-static@next
```
My resulting svelte.config.js file looked like 
```
/** @type {import('@sveltejs/kit').Config} */
import adapter from '@sveltejs/adapter-static';

export default {
	kit: {
    target: '#svelte',
		adapter: adapter({
			pages: 'build',
			assets: 'build',
			fallback: null
		}),
    vite: {
      resolve: {
        alias: {
          './runtimeConfig': './runtimeConfig.browser',
        },
      },
    },
	}
};
```
 `alias: { './runtimeConfig': './runtimeConfig.browser', }` part had to be done for it to successfully build with amplify

Now phew got through this stuff. `npm run build` creates build folder successfully `npm run preview` runs app successfully. Ok push to github and in admin ui add the front end to existing backend and fails. Then was looking into using the amplify.yml to make it potentially work and this is where landed though still didn't work

```
version: 1
env:
  variables:
    USER_BRANCH: staging
backend:
  phases:
    build:
      commands:
        - '# Execute Amplify CLI with the helper script'
        - amplifyPush --simple
frontend:
  phases:
    preBuild:
      commands:
        - npm install
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: build
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```
Error in amplify build process is long but important part assuming is 
```
2021-11-15T22:27:51.722Z [WARNING]: ✔ Successfully pulled backend environment staging from the cloud.
2021-11-15T22:27:51.745Z [WARNING]: ✖ There was an error initializing your environment.
```
Then I saw Rich Harris the main svelte guy was going to be working on svelte and be backed by vercel so I was like I have no idea what that is but if they like sveltekit then lets check it out. And they did hosting so tried that and successfully got sveltekit projects working however not with my amplify backend and wouldnnt be able to find ../aws-exports.js which is because it was in gitignore which explicitly says dont edit this and I'm not super well versed on the exact proceedures with that so that didnt work and josef in the discord group who seems pretty smart and helpful said I could look into doing a bash script but he hasnt tried that and not sure if it will work. I dont even know what a bash script is so kindof just stuck hoping some sveltekit community beaties somehow pull through and eventually a person smarter than me figures this out. 

This is my kindof base project was trying to use 
https://github.com/24jr/agility

**Comment by ashika01** [comment]
@24jr Thanks for the very detailed summary. This is super useful. To support svelte, it would be cross team effort and some help from SDK side as well. We have this discussion in our UI framework - https://github.com/aws-amplify/amplify-ui/discussions/469

I am going to move this issue to UI since they will be helping with the framework and if there are bits that needs to be implemented on JS and Console we will work with them to get this one.
